packages/amplify_datastore/example/integration_test/utils/setup_utils.dart

```diff
@@ -34,6 +56,45 @@ Future<void> configureDataStore() async {
 ///
 /// see: https:///github.com/aws-amplify/amplify-android/issues/1464
 Future<void> clearDataStore() async {
-  await Future.delayed(Duration(milliseconds: 100));
+  await Future.delayed(Duration(seconds: 2));
   await Amplify.DataStore.clear();
 }
+
+/// Am async operator that starts DataStore API sync.
+/// It returns a Future which complets when [Amplify.Hub] receives the ready
+/// event on [HubChannel.DataStore], or completes with an timeout error if
+/// the ready event hasn't been emitted with in 2 minutes.
+class DataStoreStarter {
+  final Completer _completer = Completer();
+  late StreamSubscription hubSubscription;
+
+  Future<void> startDataStore() async {
+    hubSubscription = Amplify.Hub.listen([HubChannel.DataStore], (event) {
+      if (event.eventName == 'ready') {
+        print(
+            '🎉🎉🎉DataStore is ready to start running test suites with API sync enabled.🎉🎉🎉');
+        hubSubscription.cancel();
+        _completer.complete();
+      }
+    });
+    startTimeout();
+    await Amplify.DataStore.start();
```

Comment:
Could this be changed to `await Amplify.DataStore.start().timeout(Duration(minutes: 2));` and remove the previous line and timeout util?

Reply:
The timeout is determined by how long it takes to receive the `ready` event from DataStore hub.
Using `timeout` to alter `Amplify.DataStore.start()` returned future is inaccurate I think...

Thinking of this point, `Amplify.DataStore.start()` is not need to be `await`'ed I'll remove it.